### DISCUSSION TITLE
BUG:只有把应用放在前台时才有代理作用

### DISCUSSION THREAD

**Discussion by XCQ0607** [2dust/v2rayNG#3028] [General]
### 实际行为
描述实际发生了什么
就十分奇怪，只有把软件放在用户界面才能成功代理，我有两个手机，原本都使用的是旧版本，我这个手机旧版能用，之前另一个手机就有这样的情况，我没有管。
现在我把这个手机的软件更新了，结果这个手机也变成这样了，不知道是不是华为手机的原因

### 复现方法
1.手机  华为nova 5i pro
2.设置里能开的权限全开

### 额外信息（可选）

https://github.com/2dust/v2rayNG/assets/125746521/b1210154-1f51-46bf-9054-445fe931d6cc

https://github.com/2dust/v2rayNG/assets/125746521/58cb89c2-668b-48c9-baa1-8f0c2a97fb15

### 日志信息:
--------- beginning of crash
--------- beginning of system
--------- beginning of main

//关闭小窗前

04-20 08:03:32.068 I/GoLog   (12939): tcp:127.0.0.1:55476 accepted tcp:118.194.33.18:443 [socks >> proxy]
04-20 08:03:32.069 I/GoLog   (12939): tcp:127.0.0.1:55486 accepted tcp:121.36.116.177:443 [socks >> proxy]
04-20 08:03:32.114 I/GoLog   (12939): Using Prepared: 104.16.148.244
04-20 08:03:32.116 I/GoLog   (12939): Using Prepared: 104.16.148.244

//关闭小窗后

04-20 08:03:45.428 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.429 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.430 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.431 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.431 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.431 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.431 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.431 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.431 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.432 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.432 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.432 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.432 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.432 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.432 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.432 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.432 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.432 E/tun2socks(12974): ERROR(BDatagram): recv failed
04-20 08:03:45.432 E/tun2socks(12974): ERROR(BDatagram): recv failed

//恢复小窗后

04-20 08:03:47.478 I/GoLog   (12939): tcp:127.0.0.1:41856 accepted tcp:124.70.117.176:443 [socks >> proxy]
04-20 08:03:52.587 I/GoLog   (12939): Using Prepared: 104.16.148.244

**Comment by 2dust** [comment]
搜索下华为保活app吧
